chore(curriculum): user configuration manager

zairahira opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #694

<!-- Feel free to add any additional description of changes below this line -->

zairahira commented:
@Dario-DC, I have creates a new prototype that now only uses dictionaries.

zairahira commented:
@Dario-DC,

In the updated prototype, I have added CRUD operations and have also updated the user stories.
Let me know what you think.

gikf commented:
Wouldn't it be better to not rely on the global `settings` accessed from inside of each function? And to limit using `print` in these functions?

It'd make testing way easier and functions would be more flexible.

ilenia-magoni commented on fullstack-cert/python-projects/lab-confguration-manager/main.py:
```diff
@@ -0,0 +1,71 @@
+def add_setting(settings):
+    key = input("Enter setting name to add: ").lower()
```
are we able to test this? doesn't this waiting for an input block execution?

Dario-DC replied:
Maybe we should remove the `input` calls and pass in a value as a second argument. This would remove the interactive menu but it would be easier to test everything.

zairahira commented:
@Dario-DC, I have made the following changes:

- Removed input calls.
- Removed the `main()` function.
- Removed the interactive menu inside `main()`.
- Modified each of the setting functions to take a second argument - the key or value - instead of asking for user input internally.

Dario-DC commented on fullstack-cert/python-projects/lab-confguration-manager/userstories.md:
````diff
@@ -0,0 +1,53 @@
+       "notifications": "enabled"
+   ```
+
+2. You should define a function named `add_setting` that takes the `settings` dictionary and a tuple `(key, value)` as parameters to add a new setting.
+
+3. Inside the `add_setting` function, you should:
+
+   - Convert the provided key to lowercase.
+   - Check if the setting name already exists in the `settings` dictionary.
+
+     - If the setting exists, return: `"Setting '[key]' already exists! Cannot add a new setting with this name."`
+     - If the setting doesn't exist, add the key-value pair to the dictionary.
+     - Return a confirmation message in the format: `"Setting '[key]' added with value '[value]' successfully!"`
````
```suggestion
1. You should define a function named `add_setting` that takes a dictionary of settings and a tuple containing a key-value pair as arguments.
1. `add_setting(settings, (key, value))` should:
    - Convert `key` to lowercase.
    - If the `key` setting exists, return `Setting 'key' already exists! Cannot add a new setting with this name.`
    - If the `key` setting doesn't exist, add the key-value pair to the `settings` dictionary and return `Setting 'key' added with value 'value' successfully!`
```

Dario-DC commented on fullstack-cert/python-projects/lab-confguration-manager/userstories.md:
```diff
@@ -0,0 +1,53 @@
+     - If the setting doesn't exist, add the key-value pair to the dictionary.
+     - Return a confirmation message in the format: `"Setting '[key]' added with value '[value]' successfully!"`
+
+4. You should define a function named `update_setting` that takes the `settings` dictionary and a tuple `(key_value)` as parameters to update an existing setting.
+
+5. Inside the `update_setting` function, you should:
+
+   - Convert the provided key to lowercase.
+   - Check if the setting exists in the `settings` dictionary.
+
+     - If the setting exists, update its value in the dictionary and return: `"Setting '[key]' updated to '[value]' successfully!"`
+     - If the setting does not exist, return: `"Setting '[key]' does not exist! Cannot update a non-existing setting."`
```
```suggestion
1. You should define a function named `update_setting` that takes the a dictionary of settings and a tuple containing a key-value pair as arguments.
1. `update_setting(settings, (key, value))` should:
    - Convert `key` to lowercase.
    - If the `key` setting exists, update its value in the `settings` dictionary and return: `Setting 'key' updated to 'value' successfully!`
    - If the `key` setting doesn't exist, return `Setting 'key' does not exist! Cannot update a non-existing setting.`
```

Dario-DC commented on fullstack-cert/python-projects/lab-confguration-manager/userstories.md:
```diff
@@ -0,0 +1,53 @@
+     - If the setting exists, update its value in the dictionary and return: `"Setting '[key]' updated to '[value]' successfully!"`
+     - If the setting does not exist, return: `"Setting '[key]' does not exist! Cannot update a non-existing setting."`
+
+6. You should define a function named `delete_setting` that takes the `settings` dictionary and a string `key` as parameters to delete a specific setting.
+
+7. Inside the `delete_setting` function, you should:
+
+   - Convert the provided key to lowercase.
+   - Check if the setting exists in the `settings` dictionary.
+
+     - If the setting exists, remove the key-value pair and return: `"Setting '[key]' deleted successfully!"`
+     - If the setting does not exist, return: `"Setting not found!"`
```
```suggestion
1. You should define a function named `delete_setting` that takes the a dictionary of settings and a key as arguments.
1. `delete_setting(settings, key)` should:
    - Convert `key` to lowercase.
    - If the `key` setting exists, remove the key-value pair from `settings` and return `Setting 'key' deleted successfully!`
    - If the `key` setting does not exist, return `Setting not found!`
```

Dario-DC commented on fullstack-cert/python-projects/lab-confguration-manager/userstories.md:
```diff
@@ -0,0 +1,53 @@
+     - If the setting exists, remove the key-value pair and return: `"Setting '[key]' deleted successfully!"`
+     - If the setting does not exist, return: `"Setting not found!"`
+
+8. You should define a function named `view_settings` that takes the `settings` dictionary as a parameter and returns a formatted string of current settings.
+
+9. Inside the `view_settings` function, you should:
+
+   - Check if the `settings` dictionary contains values.
+   - If the dictionary is empty, return: `"No settings available."`
+   - If the dictionary contains settings, return them in the following format:
```
```suggestion
1. You should define a function named `view_settings` that takes a dictionary of settings as argument.
1. `view_settings(settings)` should:
    - If the `settings` dictionary is empty, return `No settings available.`
    - If the `settings` dictionary contains settings, return them in the following format:
```

zairahira replied:
This makes the sentence flow better:

```suggestion
1. You should define a function named `view_settings` that takes a dictionary of settings as argument.
1. `view_settings(settings)` should:
    - Return `No settings available.` if the `settings` dictionary is empty.
    - Return them in the following format if the `settings` dictionary contains settings:
```

Dario-DC replied:
"Them" is used before "settings". Maybe
> Return the settings in the following format if the `settings` dictionary contains settings:

zairahira replied:
I have updated the last 2 user stories.

Dario-DC commented on fullstack-cert/python-projects/lab-confguration-manager/userstories.md:
````diff
@@ -0,0 +1,53 @@
+   - If the dictionary contains settings, return them in the following format:
+
+   ```md
+   Current User Settings:
````
This is not exactly the format used in the prototype. There's a missing newline at the beginning. Depending on what will be tested it might be okay to leave it as is.

Dario-DC reviewed (commented):
Left a few comments to remove the nested lists
